safe_transaction_service/history/tasks.py

```diff
@@ -272,6 +263,29 @@ def process_decoded_internal_txs_task(self) -> Optional[int]:
                 logger.info("%d Safes to process", count)
 
 
+@app.shared_task(bind=True, soft_time_limit=SOFT_TIMEOUT, time_limit=LOCK_TIMEOUT)
+def process_decoded_internal_txs_for_safe_task(
+    self, safe_address: ChecksumAddress, reindex_master_copies: bool = True
+) -> Optional[int]:
+    """
+    Process decoded internal txs for one Safe. Processing decoded transactions
+    could be slow and this way multiple Safes can be processed at the same time
+
+    :param safe_address:
+    :param reindex_master_copies: Trigger auto reindexing if a problem is found
+    :return: Number of `InternalTxDecoded` processed
+    """
+    with contextlib.suppress(LockError):
+        with only_one_running_task(self, lock_name_suffix=safe_address):
+            logger.info("[%s] Start processing decoded internal txs", safe_address)
+            index_service: IndexService = IndexServiceProvider()
+            number_processed = index_service.process_address(safe_address)
```

Comment:
```suggestion
            number_processed = index_service.process_decoded_txs(safe_address)
```
process_address sound for me too generic and confusing.